plugins/lenovo-firmware/fu-plugin-lenovo-firmware.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2021 Mario Limonciello <mario.limonciello@amd.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+	fu_plugin_add_rule (plugin, FU_PLUGIN_RULE_BETTER_THAN, "bios");
+}
+
+gboolean
+fu_plugin_startup (FuPlugin *plugin, GError **error)
+{
+	g_autofree gchar *sysfsfwdir = NULL;
+	g_autofree gchar *thinklmidir = NULL;
+
+	/* already exists */
+	sysfsfwdir = fu_common_get_path (FU_PATH_KIND_SYSFSDIR_FW_ATTRIB);
+	thinklmidir = g_build_filename (sysfsfwdir, "thinklmi", NULL);
+	if (!g_file_test (thinklmidir, G_FILE_TEST_EXISTS)) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_SUPPORTED,
+				     "thinklmi not available");
+		return FALSE;
+	}
+
+	return TRUE;
+}
+
+static gboolean
+fu_plugin_lenovo_firmware_locked (GError **error)
+{
+	gsize sz = 0;
+	g_autofree gchar *buf = NULL;
+	g_autofree gchar *sysfsfwdir = NULL;
+	g_autofree gchar *thinklmi = NULL;
+
+	sysfsfwdir = fu_common_get_path (FU_PATH_KIND_SYSFSDIR_FW_ATTRIB);
+	thinklmi = g_build_filename (sysfsfwdir, "thinklmi", "attributes",
+				     "BootOrderLock", "current_value", NULL);
+
+	/* we can't check, assume not locked */
+	if (!g_file_test (thinklmi, G_FILE_TEST_EXISTS)) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_SUPPORTED,
+				     "BootOrderLock not available");
+		return FALSE;
+	}
+
+	if (!g_file_get_contents (thinklmi, &buf, &sz, error)) {
+		g_prefix_error (error, "failed to get current value: ");
+		return FALSE;
+	}
+	g_strchomp (buf);
+
+	if (g_strcmp0 (buf, "Enable") != 0) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOTHING_TO_DO,
+				     "not locked");
```

Comment:
I'm not enamoured with returning success and value like this if we don't have to. Can't we return FALSE for error, and just pass a `gboolean *enabled (out)` or something?

Reply:
sure